### DISCUSSION TITLE
Files are not created in site/ after running `mkdocs build`

### DISCUSSION THREAD

**Discussion by fredericsimard** [squidfunk/mkdocs-material#4568] [Q&A] [answered]
## Edit N°3 : files do disappear on Linux, but not on macOS

Hi,

I created a bash script that fetches the main branch from our repo, then does a few other things before moving the files to the folder where the web server is hosting the site.

After running the mkdocs build command, some files are not converted to HTML when running on a Ubuntu VM, 20.0.4 LTS. When I run the exact same script on macOS, all the files are converted. How is that possible?

Here’s the script:

```shell
cd ~/GITHUB_REPOS
git clone [THE URL OF THE GIT]
cd gtfs.org

pip install -r requirements.txt
pip install git+https://****@github.com/squidfunk/mkdocs-material-insiders.git

sh ~/GITHUB_REPOS/tempsite/scripts/fetchdata.sh
mkdocs build --clean

cp -a ~/GITHUB_REPOS/tempsite/site/. /var/www/html/realsite/
sudo chown -R www-data:www-data /var/www/html/realsite && chmod -R 775 /var/www/html/realsite

rm -R ~/GITHUB_REPOS/tempsite/
```

Thanks,
Fred

**Comment by squidfunk** [comment]
It's a good idea to debug this by removing the "few other things" you're doing to make sure that they are not interfering. We're running Insiders on Linux (GH Actions, so Ubuntu) [to deploy our docs](https://github.com/squidfunk/mkdocs-material/blob/master/.github/workflows/documentation.yml) successfully.

**Comment by fredericsimard** [comment]
I ran each line one by one and verified if files disappeared or not, the only one that deletes files is the mkdocs build --clean command. There's also this error message during the build process that appear on Linux but not on macOS:

```
WARNING  -  A relative path to 'realtime/best-practices.md' is included in the 'nav' configuration, which is not found in the documentation files
```

What's strange is that the path is correct, the file is there in docs/, so I don't get why it says that.

**Reply by squidfunk** [reply]
Thanks for the additional info. Does it also happen if you set the `theme` in `mkdocs.yml` to `mkdocs`?

``` yaml
theme:
  name: mkdocs
```

**Reply by fredericsimard** [reply]
I'll try that. Right now I'm pursuing two new paths of possible solution. The first one is perhaps one you can answer:

Running pip install -r requirements.txt installs amongst others these 3 modules:
* `mkdocs==1.3.1`
* `mkdocs-material-extensions==1.0.3`
* `pymdown-extensions==9.5`

However, after the insiders is installed, I get this message:
```
Successfully installed mkdocs-1.4.2 mkdocs-material-8.5.7+insiders.4.26.1 mkdocs-material-extensions-1.1 pymdown-extensions-9.7
```

Is insiders installing more recent versions of modules defined in requirements.txt can be a source of issues? Or should we adjust out requirements.txt.

The other path is a possible syntax error in a shell script, although I don't understand how it would work on macOS and not Linux.

**Comment by fredericsimard** [comment, marked as answer]
I think I got it. There is a shell script being called before the mkdocs command and I was calling it with sh when I should've used bash, there were syntax errors with sh. Sorry for the trouble, mkdocs or insiders weren't the cause. How many hours have I spent when the only change required was `sh` -> `./` ...
